crates/telemetry/src/log.rs

```diff
@@ -0,0 +1,18 @@
+use std::sync::OnceLock;
+
+use crate::env;
+
+/// Takes a Spin application log and emits it as a tracing event. This acts as a compatibility layer
+/// to easily get Spin app logs as events∆1 in our OTel traces.
+pub fn app_log_to_tracing_event(buf: &[u8]) {
+    static CELL: OnceLock<bool> = OnceLock::new();
+    if *CELL.get_or_init(env::app_log_to_tracing_event_disabled) {
+        return;
+    }
+
+    if let Ok(s) = std::str::from_utf8(buf) {
+        tracing::info!(app_log = s);
+    } else {
+        tracing::info!(app_log = "Application log: <non-utf8 data>");
```

Comment:
Worth including any info that might help a log reader figure out what was going on, e.g. buffer length, initial 4 bytes, etc.?

Reply:
Great question. IMO that probably isn't valuable to a user, but let me know if you feel otherwise. Below is an idea of what it looks like for a user.

<img width="1894" alt="image" src="https://github.com/fermyon/spin/assets/20216739/823f63cf-ce24-48ff-af1b-223123e9bffa">

Reply:
The reason I was thinking about this was because it might give the reader some clue as to why they are getting this junk in the precious logs.  Like is something dumping a PNG to stdout for its own mystic reasons.  But it's an edge case for sure, and could always be retrofitted if telemetry fans demand it.

Reply:
I agree that we should emit _something_ about the actual data, maybe something like:

`app_log_non_utf8 = <ascii-escaped bytes>`

or for long data:

`app_log_non_utf8 = "<first 50 bytes, ascii-escaped> [truncated; total size = 1000]"`

https://doc.rust-lang.org/std/ascii/fn.escape_default.html